src/cli/domain/package-components.js

```diff
@@ -2,131 +2,57 @@
-      return callback(new Error('name not valid'));
-    }
-
-    async.waterfall(
-      [
-        function(cb) {
-          // Packaging template.js
+    fs.emptyDirSync(publishPath);
 
-          packageTemplate(
-            {
-              componentName: component.name,
-              componentPath: componentPath,
-              ocOptions: component.oc,
-              publishPath: publishPath
-            },
-            (err, packagedTemplateInfo) => {
-              if (err) {
-                return cb(err);
-              }
+    const componentPackage = fs.readJsonSync(componentPackagePath);
 
-              component.oc.files.template = packagedTemplateInfo;
-              delete component.oc.files.client;
-              cb(err, component);
-            }
-          );
-        },
-        function(component, cb) {
-          // Packaging server.js
+    const ocPackage = fs.readJsonSync(ocPackagePath);
 
-          if (!component.oc.files.data) {
-            return cb(null, component);
-          }
-
-          packageServerScript(
-            {
-              componentPath: componentPath,
-              dependencies: component.dependencies,
-              ocOptions: component.oc,
-              publishPath: publishPath,
-              verbose: options.verbose
-            },
-            (err, packagedServerScriptInfo) => {
-              if (err) {
-                return cb(err);
-              }
-
-              component.oc.files.dataProvider = packagedServerScriptInfo;
-              delete component.oc.files.data;
-              cb(err, component);
-            }
-          );
-        },
-        function(component, cb) {
-          // Packaging package.json
-
-          component.oc.version = ocInfo.version;
-          component.oc.packaged = true;
-          component.oc.date = getUnixUtcTimestamp();
-
-          if (!component.oc.files.static) {
-            component.oc.files.static = [];
-          }
-
-          if (!_.isArray(component.oc.files.static)) {
-            component.oc.files.static = [component.oc.files.static];
-          }
+    if (!validator.validateComponentName(componentPackage.name)) {
+      return callback(new Error('name not valid'));
+    }
 
-          fs.writeJson(
-            path.join(publishPath, 'package.json'),
-            component,
-            err => {
-              cb(err, component);
-            }
-          );
-        },
-        function(component, cb) {
-          // Packaging static files
-          packageStaticFiles(
-            {
-              componentPath: componentPath,
-              publishPath: publishPath,
-              minify: minify,
-              ocOptions: component.oc
-            },
-            err => cb(err, component)
-          );
+    const type = componentPackage.oc.files.template.type;
+    const compileOptions = {
+      publishPath,
+      componentPath,
+      componentPackage,
+      ocPackage,
+      minify
+      // TODO: verbose,
```

Comment:
...so, should we start passing it through, even if the current compilers will ignore it?